notify/impl.go

```diff
@@ -62,6 +66,85 @@ func Build(confs []*config.Receiver, tmpl *template.Template) map[string]Fanout
 
 const contentTypeJSON = "application/json"
 
+// TemplateData is the data passed to notification templates.
+// End-users should not be exposed to Go's type system,
+// as this will confuse them and prevent simple things like
+// simple equality checks to fail. Map everything to float64/string.
+type TemplateData struct {
+	Status            string
+	Alerts            []TemplateAlert
+	AlertCommonLabels map[string]string
+
```

Comment:
You don't see any use for common annotations? Both suffer from the problem of changing as more alerts are sorted into the group. But if for example `summary` is common across all of them, that can be useful for some templates.